From 20c1ca3cbb5f870ef062113c98f532669253fdb3 Mon Sep 17 00:00:00 2001
From: Wangda Tan <wangda@apache.org>
Date: Tue, 20 Jan 2015 14:42:41 -0800
Subject: [PATCH 1349/2748] YARN-2731. Fixed
 RegisterApplicationMasterResponsePBImpl to
 properly invoke maybeInitBuilder. (Contributed by
 Carlo Curino)

(cherry picked from commit f250ad1773b19713d6aea81ae290ebb4c90fd44b)
(cherry picked from commit 610aa71a103f223c258a493c8969f4909eb9dc8d)

Change-Id: I0589fe9830b1f05af9fe8297bd17088627f39d1c
---
 .../RegisterApplicationMasterResponsePBImpl.java   |    3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-common/src/main/java/org/apache/hadoop/yarn/api/protocolrecords/impl/pb/RegisterApplicationMasterResponsePBImpl.java b/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-common/src/main/java/org/apache/hadoop/yarn/api/protocolrecords/impl/pb/RegisterApplicationMasterResponsePBImpl.java
index 32dc85d..a95aadf 100644
--- a/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-common/src/main/java/org/apache/hadoop/yarn/api/protocolrecords/impl/pb/RegisterApplicationMasterResponsePBImpl.java
+++ b/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-common/src/main/java/org/apache/hadoop/yarn/api/protocolrecords/impl/pb/RegisterApplicationMasterResponsePBImpl.java
@@ -229,11 +229,11 @@ public void setApplicationACLs(
   
   @Override
   public void setClientToAMTokenMasterKey(ByteBuffer key) {
+    maybeInitBuilder();
     if (key == null) {
       builder.clearClientToAmTokenMasterKey();
       return;
     }
-    maybeInitBuilder();
     builder.setClientToAmTokenMasterKey(ByteString.copyFrom(key));
   }
   
@@ -316,6 +316,7 @@ private void addContainersFromPreviousAttemptToProto() {
   
   @Override
   public void setNMTokensFromPreviousAttempts(final List<NMToken> nmTokens) {
+    maybeInitBuilder();
     if (nmTokens == null || nmTokens.isEmpty()) {
       if (this.nmTokens != null) {
         this.nmTokens.clear();
-- 
1.7.9.5

